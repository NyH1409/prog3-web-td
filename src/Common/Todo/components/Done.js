import { TaskAlt } from "@mui/icons-material";
import { Card, CardContent, Paper, Typography } from "@mui/material";
import { pink, red } from "@mui/material/colors";
import React from "react";

export function Done(props) {
    const {done, title} = props;

    return (
        <>
        <Paper elevation={3} className="done-container" sx={{ width: 275 }}>
        <Typography variant="h5" sx={{ textAlign: "center", fontWeight:"bold", color: red[300] }}>DONE</Typography>
            {
                done?.map((elt)=>(
                    <Card sx={{ width : 275, margin : "8px 0" }} elevation={5} key={elt?.id}>
                        <CardContent>
                            <TaskAlt sx={{ color: pink[500]}} />
                            <Typography data-testid="title" color="text.secondary">{title}</Typography>
                            <Typography sx={{ mb: 1.5 }} color="blue">{new Date().toISOString().slice(0, 10)}</Typography>
                            <Typography data-testid="element-title" variant="body2" sx={{paddingRight: 2}}>{elt?.title}<br /></Typography>
                        </CardContent>
                    </Card>
                ))
            }
        </Paper>
        </>
    )
}