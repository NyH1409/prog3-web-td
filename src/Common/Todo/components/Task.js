import { Receipt } from "@mui/icons-material";
import { Card, CardActions, CardContent, FormControlLabel, FormGroup, Paper, Typography } from "@mui/material";
import { pink, red } from "@mui/material/colors";
import React from "react";

export function Task(props) {
    const {data, doneTask, title} = props;
    return (
        <>
        <Paper elevation={3} className="todo-container" 
        sx={{ width: 275, minHeight: 400, bgcolor: red[300], }}>
            <Typography variant="h5" sx={{ textAlign: "center" , fontWeight: "bold", color: "white" }}>TODO</Typography>
            {
                data?.map((elt)=>(
                    <Card sx={{ width : 275, margin : "8px 0" }} elevation={5}>
                        <CardContent>
                            <Receipt sx={{ color: pink[500]}} />
                            <Typography color="text.secondary">{title}</Typography>
                            <Typography sx={{ mb: 1.5 }} color="blue">{new Date().toISOString().slice(0, 10)}</Typography>
                            <Typography variant="body2" sx={{paddingRight: 2}}>{elt?.title}<br /></Typography>
                        </CardContent>
                        <CardActions sx={{ paddingLeft: 2}}>
                            <FormGroup>
                                <FormControlLabel control={<input className="to-check" type="checkbox" onClick={()=>doneTask(elt)}/>} label="Check if done"/>
                            </FormGroup>
                        </CardActions>
                    </Card>
                ))
            }
        </Paper>
        </>
    )
}