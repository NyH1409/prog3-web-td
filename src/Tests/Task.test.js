import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Task } from "../Common/Todo/components/Task"
import { doneTask, task } from "./TestUtils/TestUtils";

describe("Test task component", ()=>{
    it("Render correctly", ()=>{
        const component = renderer.create(
            <Task data={task} title="Task title" />
        )
        const tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
        expect(task).toMatchSnapshot([{
            id: expect.any(Number),
            title: "Test title"
        }]);
        expect("Task title").toMatchSnapshot();
    })

    it("Read component props ok", ()=>{
        const { getByTestId } = render(
            <Task data={task} title="Task title" doneTask={doneTask} />
        )

        fireEvent.click(getByTestId("to-check"));

        expect(doneTask).toHaveBeenCalled();
        expect(getByTestId("title").textContent).toBe("Task title");
        expect(getByTestId("element-title").textContent).toBe("Test title");
    })
})