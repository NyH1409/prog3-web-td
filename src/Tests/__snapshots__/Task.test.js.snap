// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test task component Render correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 todo-container css-102lv6m-MuiPaper-root"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-c24xf0-MuiTypography-root"
  >
    TODO
  </h5>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 MuiCard-root css-29ol60-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1b32eso-MuiSvgIcon-root"
        data-testid="ReceiptIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 17H6v-2h12v2zm0-4H6v-2h12v2zm0-4H6V7h12v2zM3 22l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20z"
        />
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        data-testid="title"
      >
        Task title
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-16ezf3a-MuiTypography-root"
      >
        2023-02-06
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 css-14o3s5-MuiTypography-root"
        data-testid="element-title"
      >
        Test title
        <br />
      </p>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing css-311rz8-MuiCardActions-root"
    >
      <div
        className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
      >
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
        >
          <input
            className="to-check"
            data-testid="to-check"
            onClick={[Function]}
            type="checkbox"
          />
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Check if done
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test task component Render correctly 2`] = `
Array [
  Object {
    "id": Any<Number>,
    "title": "Test title",
  },
]
`;

exports[`Test task component Render correctly 3`] = `"Task title"`;
